### PR TITLE
Adds carrot shivs

### PR DESCRIPTION
**Pull request by PKPenguin321**
Made by applying any kind of knife/a botany hatchet to a carrot. They deal 8 damage on hit, 12 on throw, ~~and have the same chance to embed as a spear.~~

https://tgstation13.org/phpBB/viewtopic.php?f=9&t=5676

:cl: PKPenguin321
rscadd: You can now sharpen carrots into shivs by using a knife or hatchet on them.
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
That's really fucking strong dude.

>plant bag with 50 carrots inside
>600 damage worth of throwing weapons per bag
>embeds also

**Comment by PKPenguin321** [comment]
carrots op please nerf

**Comment by Zilenan91** [comment]
Perhaps they shouldn't embed.

**Comment by PKPenguin321** [comment]
uuuugh fine

**Comment by drynwyn** [comment]
Nah, they're fine embedding- embedding spears are XBAWKS DEADLY because they have a high w_class, and embed damage is based on w_class.

Carrots would only deal 2 damage on embedding, and 10 damage on falling out. If you think that's too much but want them to still embed, put in a line that embedded_pain_multiplier = 1 to halve that.

Hell, you could set it to 0 and it would embed, but inflict no damage for being embedded, only when removed, which would be fun.

**Comment by PKPenguin321** [comment]
a third commit specifically to undo the previous commit feels so dirty

**Comment by Zilenan91** [comment]
I really don't think they should embed, they have such a low W_Class you could carry a backpack full of  better syndi cards basically which are a traitor item for a reason. Letting non-antags have upgraded syndi-cards every round would be a bit too cheesey.

**Comment by drynwyn** [comment]
They don't stack, syndicards do. That's a huge difference. I say we add them asis and nerf them if needed.

**Comment by PKPenguin321** [comment]
**ULTRA SERIOUS CARROT BALANCING DISCUSSION**

**Comment by KazeEspada** [comment]
>wasting all your backpack space on carrots that EVERYONE SAW YOU GROW?

**Comment by octareenroon91** [comment]
Shouldn't this maybe scale on potency?

**Comment by Lumipharon** [comment]
please no, 100 potency carrots would be like insta death javalins or some shit.

Also because carrots are small items, you can still fit like 20 of them in a bag, or more practically speaking, have 5 or so in your bag/pockets, and a plant bag filled with another 50 uncut carrots + hatchet.

Embedding is seriously strong though, I would seriously suggest not having that, or having it very weak.

**Comment by KazeEspada** [comment]
You act like this is the worst thing Botany could produce? 40 damage death nettles anyone? Tomatoes that slip you for an entire minute? SPACEEE VINNNNNNNNNNESS?

**Comment by bgobandit** [comment]
how the fuck does a carrot do more damage than a crowbar

**Comment by Lumipharon** [comment]
@KazeEspada high potency shit like bananas and the like are also bullshit and grief fodder - doesn't justify adding more bullshit.

**Comment by xxalpha** [comment]
Calculate force based on the carrot's potency.

**Comment by Robertha89** [comment]
1 potency does 8 brute, 100 does 80 brute, it's the perfect plan.

**Comment by xxalpha** [comment]
1 potency does no damage, 100 potency does 8.

**Comment by octareenroon91** [comment]
>1 potency does no damage, 100 potency does 8.
This is what I was getting at.

**Comment by PKPenguin321** [comment]
@bgobandit the carrot is sharp. the crowbar is not sharp. it is science.

@Lumipharon it already doesn't embed, see the second commit

i'd rather not make it scale with potency because then literally nobody would ever use it (le deathnettle instakill)

**Comment by xxalpha** [comment]
12 throwforce is kind of excessive, though.

**Comment by PKPenguin321** [comment]
you do realize that's literally just a floor tile + 2 and floortiles are stackable up to 50 without having to clunkily them around/cut them into knives
it's not that big a deal, calm down
and even if somebody is hucking carrots at you, you can always just catch them
besides, the image of people throwing carrots at each other TO THE DEATH is something we need in our lives

**Comment by KazeEspada** [comment]
Yes, remember catching nerfed throwing weapons.

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -1034,6 +1034,15 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
 		reagents.add_reagent("oculine", 3+round(potency / 5, 1))
 		bitesize = 1 + round(reagents.total_volume / 2, 1)
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/carrot/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/hatchet))
+		user << "<span class='notice'>You sharpen the carrot into a shiv with [I].</span>"
+		var/obj/item/weapon/kitchen/knife/carrotshiv/Shiv = new /obj/item/weapon/kitchen/knife/carrotshiv
```
Comment (neersighted) [inline code/modules/hydroponics/grown.dm]: No caps var names

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -1034,6 +1034,15 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
 		reagents.add_reagent("oculine", 3+round(potency / 5, 1))
 		bitesize = 1 + round(reagents.total_volume / 2, 1)
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/carrot/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/hatchet))
+		user << "<span class='notice'>You sharpen the carrot into a shiv with [I].</span>"
+		var/obj/item/weapon/kitchen/knife/carrotshiv/Shiv = new /obj/item/weapon/kitchen/knife/carrotshiv
+		if(!remove_item_from_storage(user))
+			user.unEquip(src)
```
Comment (neersighted) [inline code/modules/hydroponics/grown.dm]:
```
if(!user.drop_item())
    return
```

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/hydroponics/grown.dm]: why not, just curious

File: code/modules/hydroponics/grown.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/modules/hydroponics/grown.dm]: apparently i make this mistake often
iirc i think it was remie that said to do both what you're saying AND unequip it, but my memory's foggy so i'll just do what you're saying

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (xxalpha) [inline code/modules/hydroponics/grown.dm]: code standard it's somewhere in the readme

File: code/modules/hydroponics/grown.dm [hunk 3]
```diff
@@ -1034,6 +1034,16 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
 		reagents.add_reagent("oculine", 3+round(potency / 5, 1))
 		bitesize = 1 + round(reagents.total_volume / 2, 1)
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/carrot/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/hatchet))
+		user << "<span class='notice'>You sharpen the carrot into a shiv with [I].</span>"
+		var/obj/item/weapon/kitchen/knife/carrotshiv/shiv = new /obj/item/weapon/kitchen/knife/carrotshiv
```
Comment (neersighted) [inline code/modules/hydroponics/grown.dm]: No need for path here, just new()

Comment (PKPenguin321) [inline code/modules/hydroponics/grown.dm]: so just 'var/obj/item/weapon/kitchen/knife/carrotshiv/shiv = new' ?

Comment (neersighted) [inline code/modules/hydroponics/grown.dm]: `var/obj/item/weapon/kitchen/knife/carrotshiv/shiv = new()`

**Comment by demonfiren** [comment]
Instead of making damage scale with potency, make embed chance scale instead.

**Comment by octareenroon91** [comment]
>you will be able to blind people with carrots following this merge

**Comment by PKPenguin321** [comment]
Actually you wont since I never actually gave them eyestabbing, but you can pretend

**Comment by KorPhaeron** [comment]
People throwing a fit about weapon force is always funny when you remember that stuns are still plentiful.

**Comment by Lumipharon** [comment]
spears aren't really dodgable compared to a taser. Also not everyone has a taser, but 20 greyshits can all easily get spears.

**Comment by Zilenan91** [comment]
You can have like eight spears on you at once too if you put wired rods and glass shards in your bag.

**Comment by PKPenguin321** [comment]
@tgstation-server merge this shit i have 5 open prs i can't handle the commitment

**Comment by PKPenguin321** [comment]
@Razharas @RemieRichards @duncathan anybody with merge access do the mergey thing PLEASE

**Comment by duncathan** [comment]
We can't

**Comment by PKPenguin321** [comment]
i bet its an inhands sprite conflict but wont be sure until i can check later

**Comment by PKPenguin321** [comment]
was gonna fix this today but git went down and now i'm busy af, will fix tomorrow probably.

**Comment by PKPenguin321** [comment]
welp

**Comment by PKPenguin321** [comment]
fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git fuck git :rage:

**Pull request closed without merging**